tests/cpp/test_gpu1.cpp

```diff
@@ -2696,27 +2698,52 @@ TEST_F(NVFuserTest, Fp8CastOps) {
 
       at::Tensor t0 = at::randn({1, 4}, options);
 
+      if (fp8_type == DataType::Float8_e8m0fnu) {
+        // e8m0 can only represent positive values
+        t0 = t0.abs() + 1e-6;
+      }
+
       KernelExecutor ke;
+      bool unsupported_device = false;
+      if (fp8_type == DataType::Float8_e8m0fnu) {
+#if (CUDA_VERSION >= 12070)
+        unsupported_device = !deviceMajorMinorCheck(10);
+#else
+        unsupported_device = true;
+#endif
+      } else {
 #if (CUDA_VERSION >= 12010)
-      if (!deviceMajorMinorCheck(8, 9)) {
+        unsupported_device = !deviceMajorMinorCheck(8, 9);
 #elif (CUDA_VERSION >= 11080)
-      if (!deviceMajorMinorCheck(9)) {
+        unsupported_device = !deviceMajorMinorCheck(9);
 #else
-      if (true) {
+        unsupported_device = true;
 #endif
+      }
+      if (unsupported_device) {
         ASSERT_THAT(
             [&]() { ke.compile(&fusion, {t0}); },
-            testing::ThrowsMessage<nvfuser::nvfError>(testing::HasSubstr(
-                "Reason: Fusion contains Float8_xxx values")));
+            testing::ThrowsMessage<nvfuser::nvfError>(
+                testing::HasSubstr("Reason: Fusion contains Float8_")));
       } else {
         ke.compile(&fusion, {t0});
         auto outputs = ke.run({t0});
 
         at::Tensor ref_output = t0.to(at_fp8_type).to(at_src_type);
 
         NVF_CHECK(
-            outputs[0].as<at::Tensor>().equal(ref_output),
-            "cast to fp8 and back had a mismatch.\n",
+            fp8_type == DataType::Float8_e8m0fnu
+                ? (outputs[0]
+                       .as<at::Tensor>()
+                       .ge(ref_output / 2)
+                       .logical_and(
+                           outputs[0].as<at::Tensor>().le(ref_output * 2))
+                       .all()
+                       .item<bool>())
```

Comment:
Looks like [PyTorch's implementation](https://github.com/pytorch/pytorch/blob/3dd872e6d53560933d8d7fc11357617746d37168/c10/util/Float8_e8m0fnu.h#L39) is different from NVIDIA's implementation...

Reply:
Could you explain what you're doing here? Looks like it accounts for a 2x error range?

Reply:
Yes exactly. e8m0 has no mantissa, so it can only represent 1*2^x, and I am asserting that, if the results are different, then x is different by at most 1.